db/python/tables/family.py

```diff
@@ -29,16 +29,32 @@ async def get_projects_by_family_ids(self, family_ids: List[int]) -> Set[Project
             )
         return projects
 
-    async def get_families(self, project: int = None) -> List[Family]:
+    async def get_families(
+        self, project: int = None, participant_ids: List[int] = None
+    ) -> List[Family]:
         """Get all families for some project"""
-        _query = """\
-SELECT id, external_id, description, coded_phenotype, project
-FROM family
-WHERE project = :project"""
+        _query = """
+            SELECT id, external_id, description, coded_phenotype, project
+            FROM family
+        """
 
-        rows = await self.connection.fetch_all(
-            _query, {'project': project or self.project}
-        )
+        where: List[str] = []
+
+        if participant_ids:
+            _query += """
+                JOIN family_participant
+                ON family.id = family_participant.family_id
+            """
+            pids = ', '.join([str(x) for x in participant_ids])
+            where.append(f'participant_id IN ({pids})')
```

Comment:
We should use SQL placeholders here to mitigate risk of SQL injection attacks, it's not practical given the current structure of code, but this could be changed and forgotten this is constructing the values directly.

Reply:
Ah ok makes sense. I can convert it.

Reply:
The SQL engine we're using is clever enough that it knows arrays, so you can just do something like

```
values_to_replace = {}
if participant_ids:
    where.append(f'participant_id IN :pids')
    values_to_replace['pids'] = participant_ids
# ....
connection.execute(_query, values)


```